io-sim/src/Control/Monad/IOSim.hs

```diff
@@ -830,12 +853,23 @@ schedule thread@Thread{
       trace <- schedule thread' simstate { timers = timers' }
       return (Trace time tid tlbl (EventTimerUpdated tmid expiry) trace)
 
+    -- updating a negative timer is a no-op, unlike in `GHC.Event`.
+    UpdateTimeout (NegativeTimeout _tmid) _d k -> do
```

Comment:
And a good reason to distinguish a `NegativeTimeout` constructor. So yes I agree.